cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/Python.kt

```diff
@@ -135,15 +135,15 @@ interface Python {
      *  |  | Continue
      * ```
      */
-    abstract class ASTBASEstmt(pyObject: PyObject) : AST(pyObject), WithPythonLocation
+    sealed class ASTBASEstmt(pyObject: PyObject) : AST(pyObject), WithPythonLocation
 
     /**
      * ```
      * ast.FunctionDef = class FunctionDef(stmt)
      *  |  FunctionDef(identifier name, arguments args, stmt* body, expr* decorator_list, expr? returns, string? type_comment)
      * ```
      */
-    class ASTFunctionDef(pyObject: PyObject) : ASTBASEstmt(pyObject) {
+    open class ASTFunctionDef(pyObject: PyObject) : ASTBASEstmt(pyObject) {
```

Comment:
The main point of making things open here was that some classes are mainly a more specific implementation of another class, so I thought it's more elegant to let them inherit from each other instead of copy-pasting essentially the whole code (of the translation to the kotlin-AST but also in the frontend).

Reply:
I changed `open` to `sealed`, does that make it better @maximiliankaul ?